Sources/Node/Convertible/Date+Convertible.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+extension Date: NodeConvertible {
+    
+    static fileprivate let dateFormatter = { () -> DateFormatter in
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss.ssssss"
+        return dateFormatter
+    }()
+    
+    /**
+     Turn the Date into a node
+     
+     - throws: if convertible can not create a Node
+     - returns: a node if possible
+     */
+    public func makeNode(context: Context) throws -> Node {
+        let string = Date.dateFormatter.string(from:self)
+        
+        return Node(string)
+    }
+    
+    /**
+     Initialize the Date with a node within a context.
+     
+     Context is an empty protocol to which any type can conform.
+     This allows flexibility. for objects that might require access
+     to a context outside of the node ecosystem
+     */
+    public init(node: Node, in context: Context) throws {
+        
+        switch (node) {
+        case .string (let string):
```

Comment:
I'm actually no longer supportive of PR as written.  In order to support sub-second time report, I had to change the string format to include "HH:mm:ss.sssssss".  But this then breaks compliance with standard JSON timestamps, since the same conversion will be used from JSON to node and node to MySQL.  

I think it makes sense to support Double (since it can represent all time formats) but right now the MySQL code cannot use that since it does not have enough hints when preparing a statement to choose to convert the Double into a MYSQL-appropriate format.